corehq/apps/hqwebapp/static/hqwebapp/js/hqModules.js

```diff
@@ -0,0 +1,24 @@
+var COMMCAREHQ_MODULES = {};
+
+function hqDefine(path, moduleObject) {
```

Comment:
How hard would it be make this signature match requirejs' `define` a bit more closely?

```js
function hqDefine(path, moduleFunc) {
```
`hqDefine` should call `moduleFunc` and the returned object is the module object. Alternately, we could make it take a function or object, and if it's a function call it and use the returned object, otherwise use the object as the module.